### PR TITLE
Fix/test executable path

### PR DESCRIPTION
**Pull request by h1day**
This pull request enhances how the test execution working directory is determined and improves related documentation and test coverage. The main change is that the `testExecutablePath` setting now controls the working directory for test execution, with support for variable substitution (such as `${workspaceFolder}`), and this behavior is now clearly documented and thoroughly tested.

**Working Directory Handling Improvements**
* The `CppUTestAdapter` now sets the test execution working directory based on the `testExecutablePath` setting, defaulting to the executable's directory if not set. This ensures more predictable and configurable test runs.

**Documentation Updates**
* The `README.md` is updated to clarify how the `testExecutablePath` setting affects the working directory for test execution, including variable support and fallback behavior.

**Test Coverage Enhancements**
* New tests in `ExecutableRunner.spec.ts` verify that the working directory is correctly set according to the provided options or defaults to the executable's directory.
* Additional tests in `SettingsProvider.spec.ts` confirm that variable substitution in `testExecutablePath` works as expected and that edge cases (empty string, no variables) are handled properly.

### CODE REVIEW

File: src/adapter.ts [hunk 1]
```diff
@@ -145,9 +145,10 @@ export class CppUTestAdapter implements TestAdapter {
 	}
 
 	private GetExecutionOptions(): ExecutableRunnerOptions | undefined {
+		const testPath = this.settingsProvider.GetTestPath();
 		return {
 			objDumpExecutable: this.settingsProvider.GetObjDumpPath(),
-			workingDirectory: undefined // this is calculated inside the ExecutableRunner, not really good style
+			workingDirectory: testPath || undefined
```
[the remark this answers is not in the document]
Comment (h1day) [reply src/adapter.ts]: @bneumann 

I’m sorry...

I shouldn’t have accepted Copilot’s suggestion. Still, since I implemented it, the responsibility is mine.

In ExecutableRunner, the code this.workingDirectory = options?.workingDirectory ?? dirname(command); uses the nullish coalescing operator (??).

With ??, the fallback is applied only when the value is null or undefined.

An empty string '' is “falsy” but not null or undefined, so it passes through.

As a result, if options.workingDirectory = '', then workingDirectory will be set to an empty string.

[the remark this answers is not in the document]
Comment (bneumann) [reply src/adapter.ts]: Oh, you are right! Nut no worry, I didn't see it too :) Only that you mention it now I think I stumbled over this problem once and that thing was `undefined` in the first place. 
I think the way you do it is better but we need to handle the string correctly later.

[the remark this answers is not in the document]
Comment (h1day) [reply src/adapter.ts]: @bneumann 

Thanks! That makes sense.
I’ll adjust it so that only a non-empty string is taken, and otherwise it falls back to dirname(command). This should cover the '' / whitespace case and keep the behavior consistent.
Does that sound good to you, or do you think we should allow explicit `''` as “use CWD”?
